plugin/js/parsers/JadeScrollsParser.js

```diff
@@ -0,0 +1,443 @@
+
+  3. URL structure: /novel/{novel-slug}/{chapter-slug}
+     Chapter slugs can be: episode-1, episode-2, chapter-1, chapter-217, etc.
+     Example: /novel/how-to-survive-as-the-second-son-of-a-mage-family/episode-1
+
+  4. Features:
+     - Fetches chapter list from API with proper sorting by episode/chapter number
+     - Retrieves chapter content directly from JSON API
+     - Builds HTML DOM from API response
+     - Caches novel metadata across parser instances using static variables
+     - Extracts metadata (title, author, cover, description, genre) from API
+     - Falls back to URL-based title extraction if metadata not loaded yet
+     - 1.5-second throttle to prevent rate limiting
+
+  5. Notes:
+     - Chapter content is in JSON field: data.content (HTML format)
+     - Novel metadata is in: data.title, data.OriginalNovelAuthor, data.coverImg, etc.
+     - Metadata is cached by novel slug for efficient access across instances
+     - If API structure changes, update buildDomFromChapterData() and getChapterUrls() methods
+*/
+"use strict";
+
+parserFactory.register("jadescrolls.com", () => new JadeScrollsParser());
+
+class JadeScrollsParser extends Parser {
+    // Static cache for metadata (shared across all instances)
+    static metadataCache = {};
+    static metadataPromises = {};
+
+    constructor() {
+        super();
+        // JadeScrolls uses API endpoints which can handle faster requests
+        // Reduced from 3000ms based on testing
+        this.minimumThrottle = 1500;
+    }
+
+    // Override getPageDOM to start metadata loading as early as possible
+    async getPageDOM(url) {
+        let dom = await super.getPageDOM(url);
+        // Start loading metadata in the background (don't await)
+        this.startMetadataLoad(dom);
+        return dom;
+    }
+
+    // Start metadata loading without waiting (call this early)
+    startMetadataLoad(dom) {
+        let novelSlug = this.extractNovelSlug(dom.baseURI);
+        if (!novelSlug) {
+            return;
+        }
+
+        // Check if already loaded or loading
+        if (JadeScrollsParser.metadataCache[novelSlug] || JadeScrollsParser.metadataPromises[novelSlug]) {
+            return;
+        }
+
+        // Start the fetch but don't await it - store the promise
+        JadeScrollsParser.metadataPromises[novelSlug] = (async () => {
+            try {
+                let apiUrl = `https://api.jadescrolls.com/api/public/get-novel-by-slug?slug=${novelSlug}&chapterSort=ASC`;
+                let jsonResponse = await HttpClient.fetchJson(apiUrl);
+                let novelData = await jsonResponse.json;
+                JadeScrollsParser.metadataCache[novelSlug] = novelData.data;
+            } catch (error) {
+                console.error("Failed to fetch novel metadata:", error);
+            }
+        })();
+    }
+
+    // Override populateUI to ensure metadata is loaded before populating the UI
+    async populateUI(dom) {
```

Comment:
base version of populateUI isn't async.  (However, probably doesn't matter as caller is async.)